src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1382,7 +1380,11 @@ torch::Tensor VideoDecoder::convertFrameToTensorUsingFilterGraph(
   ffmpegStatus =
       av_buffersink_get_frame(filterState.sinkContext, filteredFrame.get());
   TORCH_CHECK_EQ(filteredFrame->format, AV_PIX_FMT_RGB24);
-  std::vector<int64_t> shape = {filteredFrame->height, filteredFrame->width, 3};
+  int height = 0, width = 0;
+  std::tie(height, width) = getHeightAndWidthFromOptionsOrAVFrame(
+      streams_[streamIndex].options, filteredFrame.get());
+  std::vector<int64_t> shape = {height, width, 3};
```

Comment:
So this is the only place where the logic is slightly changed (but I think the behavior is the same): we go through `options` first. I *think* this is correct, because the `filteredFrame` should respect `options` itself?
In any case, this is what is done for the swscale case as well. If that's incorrect or a change of behavior, LMK.

Reply:
Looks correct to me, but @ahmadsharif1 should also reason through it.

Reply:
Maybe add a TORCH_CHECK to make sure the filteredFrame has the expected dimensions?

Reply:
I'll add the validity checks as follow up!